osquery/tables/system/darwin/quarantined_history.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// Path (after /Users/foo) where the quarantine events DB will be found
+const std::string kQuarantineEventsDbPath =
+    "Library/Preferences/com.apple.LaunchServices.QuarantineEventsV2";
+
+const int cocoaToUnixEpochOffset = 978307200;
+
+void genQuarantineEventRow(sqlite3_stmt* const stmt,
+                           std::string username,
+                           Row& r) {
+  for (int i = 0; i < sqlite3_column_count(stmt); i++) {
+    auto column_name = std::string(sqlite3_column_name(stmt, i));
+    auto column_type = sqlite3_column_type(stmt, i);
+    if (column_type == SQLITE_TEXT) {
+      auto value = sqlite3_column_text(stmt, i);
+      if (value != nullptr) {
+        r[column_name] = std::string(reinterpret_cast<const char*>(value));
+      }
+    } else if (column_type == SQLITE_FLOAT) {
+      auto value = sqlite3_column_double(stmt, i);
+      // convert to UNIX epoch
+      if (column_name == "timestamp") {
+        auto timestamp = value + cocoaToUnixEpochOffset;
+        r[column_name] = DOUBLE(timestamp);
+      } else {
+        r[column_name] = DOUBLE(value);
+      }
+    } else if (column_type == SQLITE_INTEGER) {
+      auto value = sqlite3_column_int(stmt, i);
+      r[column_name] = INTEGER(value);
+    }
+  }
+  r["username"] = TEXT(username);
+}
+
+void genQuarantineHistoryItems(const fs::path& qepath,
+                               std::string username,
+                               QueryData& results) {
+  sqlite3* db = nullptr;
+
+  try {
```

Comment:
Can you replace this `try` with a check for the `Status` using: `pathExists`: https://github.com/facebook/osquery/blob/master/include/osquery/filesystem.h#L134